object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -163,47 +169,80 @@ boolean returnsThis(final MethodInvocationTree tree) {
 
   /**
    * Given a tree, returns the method that the tree should be considered as calling. Returns
-   * "withOwners" if the call is {@code withFilters(..., new Filter("owner"), ...)}. Otherwise,
-   * returns its first argument.
+   * "withOwners" if the call sets an "owner", "owner-alias", or "owner-id" filter. Returns
+   * "withImageIds" if the call sets an "image-ids" filter.
    *
    * <p>Package-private to permit calls from {@link ObjectConstructionTransfer}.
    *
-   * @return either the first argument, or "withOwners" if the call is {@code withFilters(..., new
-   *     Filter("owner"), ...)}
+   * @return either the first argument, or "withOwners" or "withImageIds" if the tree is an
+   *     equivalent filter addition.
    */
   String adjustMethodNameUsingValueChecker(
       final String methodName, final MethodInvocationTree tree) {
-    if (useValueChecker) {
-      if ("withFilters".equals(methodName)) {
-        for (Tree filterTree : tree.getArguments()) {
-          // Search the arguments to withFilters for a Filter constructor invocation,
-          // passing through as many method invocation trees as needed. This code is searching
-          // for code of the form:
-          // new Filter("owner").withValues("...")
-          while (filterTree != null && filterTree.getKind() == Tree.Kind.METHOD_INVOCATION) {
-            filterTree =
-                TreeUtils.getReceiverTree(((MethodInvocationTree) filterTree).getMethodSelect());
+    if (!useValueChecker) {
+      return methodName;
+    }
+
+    ExecutableElement invokedMethod = TreeUtils.elementFromUse(tree);
+    if (!"com.amazonaws.services.ec2.model.DescribeImagesRequest"
+        .equals(ElementUtils.enclosingClass(invokedMethod).getQualifiedName().toString())) {
+      return methodName;
+    }
+
+    if ("withFilters".equals(methodName) || "setFilters".equals(methodName)) {
+      for (Tree filterTree : tree.getArguments()) {
+        // Search the arguments to withFilters for a Filter constructor invocation,
+        // passing through as many method invocation trees as needed. This code is searching
+        // for code of the form:
+        // new Filter("owner").withValues("...")
+        // or code of the form:
+        // new Filter().*.withName("owner").*
+
+        // Set to non-null iff a call to withName was observed; in that case, this variable's
+        // value is the argument to withName.
+        String withNameArg = null;
+        ValueAnnotatedTypeFactory valueATF = getTypeFactoryOfSubchecker(ValueChecker.class);
```

Comment:
I wonder if this would be cleaner if written as a visitor, with some top-level checks on the type of the expression before applying the visitor?  Not urgent, possibly for a future refactoring

Reply:
Maybe. I think the complexity here is still manageable, but if we were to make it much more complex I would agree with you.

For now, though, I'm going to leave it as-is.